tools/scylla-sstable.cc

```diff
@@ -912,6 +914,94 @@ void consume_sstables(schema_ptr schema, reader_permit permit, std::vector<sstab
     }
 }
 
+class scylla_sstable_table_state : public compaction::table_state {
```

Comment:
Nice to see a benefit from this abstraction: We should do it more: module X's users implement interfaces exposed by module X in order to integrate with it, rather than X knowing about the users. @raphaelsc

Reply:
Yes, these interfaces are great. This was significantly simpler than instantiating a `table` and all the boilerplate that comes with that.

Reply:
Very happy that it was useful here.